lightning-invoice/src/payment.rs

```diff
@@ -314,10 +275,18 @@ pub trait Payer {
 /// A trait defining behavior for routing an [`Invoice`] payment.
 pub trait Router {
 	/// Finds a [`Route`] between `payer` and `payee` for a payment with the given values.
-	fn find_route<S: Score>(
+	fn find_route(
 		&self, payer: &PublicKey, route_params: &RouteParameters, payment_hash: &PaymentHash,
-		first_hops: Option<&[&ChannelDetails]>, scorer: &S
+		first_hops: Option<&[&ChannelDetails]>, inflight_htlc_map: HashMap<(u64, bool), u64>
 	) -> Result<Route, LightningError>;
+	/// Lets the router know that payment through a specific path has failed.
+	fn payment_path_failed(&self, path: Vec<&RouteHop>, short_channel_id: u64);
```

Comment:
Since the context for these methods might not entirely be clear at this point and a user might not be sure when to use them, could you mention in the comments that they will update the internal scorer? Also, maybe we might want to prefix them with `notify_` at this point?

Reply:
`Router`s will not necessarily have an internal scorer unless they use `DefaultRouter.` 

Perhaps we could mention that this method is designed for delegating path failure events to an (optional) scorer?

Reply:
Fixed in ea84acfe